### PR TITLE
feat(TableTable): paste in multiple cells

### LINKED ISSUE #1615: FEAT: Paste TSV data into new row
**Description:**
When pasting data from Excel or Google Sheets to add to a PostgreSQL DB table view visually, data can only be pasted in one cell at a time; if I try to paste in multiple cells within a row, the data appears in a single cell with tabs separating the data that should be entered across multiple cells. Similarly, pasting data that should appear in multiple rows within one column all appears in a single cell.

**Alternatives:**
I haven't discovered any alternatives/workarounds for this issue.

**Additional context**
Add any other context or screenshots about the feature request here. (we love screenshots with red circles if the request is for something visual)
For example, copying the below data:
image

Leads to the following entry on selecting paste (Ctrl+V):
image

Instead of the expected behavior:
image

Not sure if this is database type specific or not. Thank you!

### PR DESCRIPTION
**Pull request by azmy60**
close #1615 

Animation4

### CODE REVIEW

File: apps/studio/src/lib/menu/tableMenu.ts [hunk 1]
```diff
@@ -130,6 +130,53 @@ export async function copyRange(options: {
...
+        setCellValue(cell, text);
+      })
+    })
+  }
+}
+
+export function setCellValue(cell: Tabulator.CellComponent, value: string) {
+  /* eslint-disable-next-line @typescript-eslint/ban-ts-comment */
+  // @ts-ignore
+  const editableFunc = cell.getColumn().getDefinition().__spreadsheet_editable
```
Comment (rathboma) [inline apps/studio/src/lib/menu/tableMenu.ts:175]: yuck, can we add an API for this on the tabulator side?

File: apps/studio/src/lib/menu/tableMenu.ts [hunk 2]
```diff
@@ -130,6 +130,53 @@ export async function copyRange(options: {
...
+  if (parsedText.errors.length > 0) {
+    const cell = range.getCells()[0]
+    setCellValue(cell, text);
+  } else {
+    const table = range.getCells()[0].getTable()
+    const rows = table.modules.spreadsheet.getRows().slice(range.getTop())
+    /* eslint-disable-next-line @typescript-eslint/ban-ts-comment */
+    // @ts-ignore
+    // TODO: use range.getLeft(), which is not available in tabulator yet
+    const columns = table.modules.spreadsheet.getColumns().slice(range._range.left + 1)
```
Comment (rathboma) [inline apps/studio/src/lib/menu/tableMenu.ts:151]: Is this part of the PR on the tabulator side?

**Review by rathboma — APPROVED** [review]
Looks great -- let's be sure to solve those gross tabulator API calls properly on the tabulator side though

File: apps/studio/src/lib/menu/tableMenu.ts (hunk 2 above)
Comment (azmy60) [reply apps/studio/src/lib/menu/tableMenu.ts:151]: Yeah. It's an easy fix.

File: apps/studio/src/lib/menu/tableMenu.ts (hunk 1 above)
Comment (azmy60) [reply apps/studio/src/lib/menu/tableMenu.ts:175]: This is the part where Oli wanted to discuss lol. Column definitions are being mutated by the spreadsheet module and we shouldn't do that. I think he said there were some use cases to fix it so I'll take a look.

**Pull request merged**